simulators/ethereum/consensus/testmodel.py

```diff
@@ -122,31 +143,41 @@ def get(self, key):
     def genesis(self, key = None):
         """ Returns the 'genesis' block for this testcase,
         including any alloc's (prestate) required """
+
+
         # Genesis block
         if self.raw_genesis is None:
+
+            #We must fix some fields in the genesis, geth is picky about that
+            fields_to_fix = ['nonce','coinbase','hash','mixHash','parentHash','receiptTrie','stateRoot','transactionsTrie','uncleHash']
+        
+
             raw_genesis = self.data['genesisBlockHeader']
 
-            # Turns out the testcases have noncewritten as 0102030405060708.
-            # Which is supposed to be interpreted as 0x0102030405060708.
-            # But if it's written as 0102030405060708 in the genesis file,
-            # it's interpreted differently. So we'll need to mod that on the fly
-            # for every testcase.
-            if 'nonce' in raw_genesis:
-                nonce = raw_genesis[u'nonce']
-                if not raw_genesis[u'nonce'][:2] == '0x':
-                    raw_genesis[u'nonce'] = '0x'+raw_genesis[u'nonce']
-
-            # Also, testcases have 'code' written as 0xdead
-            # But geth does not handle that, so we'll need to mod any of those also
-            # However, cpp-ethereum rejects 'code' written as 'dead'
-#            for addr, account in self.data['pre'].items():
-#                if 'code' in account:
-#                    code = account['code']
-#                    if code[:2] == '0x':
-#                        account['code'] = code[2:]
-
-
-            raw_genesis['alloc'] = self.data['pre']
+            for key in fields_to_fix:
+                v = raw_genesis[key]
+                if len(v) > 2 and v[:2] != '0x':
+                    raw_genesis[key] = '0x'+raw_genesis[key]
+
+            # And fix the alloc-section
+            alloc = {}
+
+            for addr in self.data['pre']:
+                v = self.data['pre'][addr]
+                if not addr[:2] == '0x':
+                    addr = '0x'+addr
```

Comment:
Because of this change we can no longer run the consensus tests against pyethereum as now the addresses end up being 42 bytes long and that makes pyethereum [think they don't need to be hex decoded](https://github.com/ethereum/pyethereum/blob/develop/ethereum/blocks.py#L1465-L1467). I suppose this needs to be fixed in pyethereum, but in the meantime it'd be nice if we could workaround it here. Would that be possible?